versioned_hdf5/tests/test_subchunk_map.py

```diff
@@ -1,136 +1,135 @@
-from collections import defaultdict
+from __future__ import annotations
+
+from typing import Any
 
 import hypothesis
-import ndindex
 import numpy as np
 import pytest
 from hypothesis import given
 from hypothesis import strategies as st
 from hypothesis.extra import numpy as stnp
-from numpy.testing import assert_equal
+from numpy.testing import assert_array_equal
 
 from ..subchunk_map import as_subchunk_map
 
+max_examples = 10_000
+
 
-def non_negative_step_slices(size):
-    start = st.one_of(st.integers(min_value=-size, max_value=size - 1), st.none())
-    stop = st.one_of(st.integers(min_value=-size, max_value=size), st.none())
+def non_negative_step_slices(size: int):
+    start = st.one_of(st.integers(-size - 1, size + 1), st.none())
+    stop = st.one_of(st.integers(-size - 1, size + 1), st.none())
     # only non-negative steps (or None) are allowed
-    step = st.one_of(st.integers(min_value=1, max_value=size), st.none())
+    step = st.one_of(st.integers(1, size + 1), st.none())
     return st.builds(slice, start, stop, step)
 
 
-@pytest.mark.slow
-@given(st.data())
-@hypothesis.settings(database=None, max_examples=10_000, deadline=None)
-def test_as_subchunk_map(data):
-    ndim = data.draw(st.integers(1, 4), label="ndim")
-    shape = data.draw(st.tuples(*[st.integers(1, 100)] * ndim), label="shape")
-    chunks = data.draw(st.tuples(*[st.integers(5, 20)] * ndim), label="chunks")
-    idx = ndindex.Tuple(
-        *[
-            data.draw(non_negative_step_slices(shape[dim]), label=f"idx{dim}")
-            for dim in range(ndim)
-        ]
+@st.composite
+def basic_idx_st(draw, shape: tuple[int, ...]) -> Any:
+    """Hypothesis draw of slice and integer indexes"""
+    nidx = draw(st.integers(0, len(shape)))
+    idx_st = st.tuples(
+        *(
+            # FIXME we should push the scalar use case into non_negative_step_slices
+            # However ndindex fails when mixing scalars and slices and array indices
+            # https://github.com/Quansight-Labs/ndindex/issues/188
```

Comment:
Shall we make an issue so that we can track this work?

Reply:
on  https://github.com/Quansight-Labs/ndindex/issues/188 @asmeurer and I reached consensus that this is not worth pursuing